### PR TITLE
Add camel-based triplestore indexing service

### PR DESCRIPTION
**Pull request by acoburn**
This resolves https://jira.duraspace.org/browse/FCREPO-1466

This also normalizes the maven configuration across projects and expands the various README files, where appropriate.

Additionally, the use of logging was made more consistent across projects.

### CODE REVIEW

File: indexing-triplestore/pom.xml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <defaultGoal>install</defaultGoal>
+
+    <plugins>
+      <plugin>
```
Comment (awoods) [inline indexing-triplestore/pom.xml]: Add:
`
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
      </plugin>
`

File: indexing-triplestore/src/test/java/org/fcrepo/camel/indexing/triplestore/integration/RouteDeleteIT.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    @Produce(uri = "direct:start")
+    protected ProducerTemplate template;
+
+    @Override
+    protected void doPreSetup() throws Exception {
+    }
+
+    @Before
+    public void setUpFuseki() throws Exception {
+        final String fusekiBase = "http://localhost:" + FUSEKI_PORT + "/test";
```
Comment (awoods) [inline indexing-triplestore/src/test/java/org/fcrepo/camel/indexing/triplestore/integration/RouteDeleteIT.java]: `fusekiBase` does not appear to be used.

File: indexing-triplestore/src/test/java/org/fcrepo/camel/indexing/triplestore/integration/RouteDeleteIT.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        getMockEndpoint(fcrepoEndpoint).expectedMessageCount(0);
+       
+        template.sendBodyAndHeaders("direct:start", "", headers);
+        
+        await().until(TestUtils.triplestoreCount(fusekiBase, fullPath), equalTo(0));
+
+        assertMockEndpointsSatisfied();
+    }
+
+    //@Test
```
Comment (awoods) [inline indexing-triplestore/src/test/java/org/fcrepo/camel/indexing/triplestore/integration/RouteDeleteIT.java]: Should a `TODO` be added for this test?

File: indexing-triplestore/pom.xml (hunk 1 above)
Comment (acoburn) [inline indexing-triplestore/pom.xml]: I also added this plugin to the indexing-solr project

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4-labs/fcrepo-camel-toolbox/commit/38d078360037a0dfd6ed05dcde08c6186a16c69c

**Pull request closed without merging**